gymnasium/envs/mujoco/utils.py

```diff
@@ -0,0 +1,64 @@
+"""A set of MujocoEnv related utilities, mainly for testing purposes.
+
+Author: @Kallinteris-Andreas
+"""
+
+import mujoco
+import numpy as np
+
+import gymnasium
+
+
+def get_state(
+    env: gymnasium.envs.mujoco.MujocoEnv,
+    state_type: mujoco.mjtState = mujoco.mjtState.mjSTATE_PHYSICS,
+):
+    """Gets the state of `env`.
+
+    Arguments:
+        env: Environment whose state to copy, `env.model` & `env.data` must be accessible.
+        state_type: see the [documentation of mjtState](https://mujoco.readthedocs.io/en/stable/APIreference/APItypes.html#mjtstate) most users can use the default for training purposes or `mujoco.mjtState.mjSTATE_INTEGRATION` for validation purposes.
+    """
+    assert mujoco.__version__ >= "2.3.6", "Feature requires `mujuco>=2.3.6`"
+
+    state = np.empty(mujoco.mj_stateSize(env.unwrapped.model, state_type))
+    mujoco.mj_getState(env.unwrapped.model, env.unwrapped.data, state, state_type)
+    return state
+
+
+def set_state(
+    env: gymnasium.envs.mujoco.MujocoEnv,
+    state: np.ndarray,
+    state_type: mujoco.mjtState = mujoco.mjtState.mjSTATE_PHYSICS,
+):
+    """Set the state of `env`.
+
+    Arguments:
+        env: Environment whose state to set, `env.model` & `env.data` must be accessible.
+        state: State to set (generated from get_state).
+        state_type: see the [documentation of mjtState](https://mujoco.readthedocs.io/en/stable/APIreference/APItypes.html#mjtstate) most users can use the default for training purposes or `mujoco.mjtState.mjSTATE_INTEGRATION` for validation purposes.
+    """
+    assert mujoco.__version__ >= "2.3.6", "Feature requires `mujuco>=2.3.6`"
+
+    mujoco.mj_setState(
+        env.unwrapped.model,
+        env.unwrapped.data,
+        state,
+        spec=mujoco.mjtState.mjSTATE_PHYSICS,
+    )
+    return state
+
+
+def check_mujoco_reset_state(env: gymnasium.envs.mujoco.MujocoEnv, seed=1234):
```

Comment:
Why is this in `mujoco/utils`

Reply:
I wanted to place somewhere `gymnasium/*` and not `tests/*` in order for it to be accessible to other libraries that `mujoco`